designcompose/src/main/java/com/android/designcompose/squoosh/SquooshText.kt

```diff
@@ -140,157 +140,136 @@ internal fun squooshComputeTextInfo(
             val builder = AnnotatedString.Builder()
             builder.append(normalizeNewlines(customizedText))
             builder.toAnnotatedString()
-        } else
-            when (val vDatatype = v.data.getType()) {
-                is ViewDataType.Text -> {
-                    val builder = AnnotatedString.Builder()
-                    builder.append(
-                        normalizeNewlines(
-                            getTextContent(appContext, vDatatype as ViewDataType.Text)
-                        )
+        } else {
+            if (v.data.hasText()) {
+                val builder = AnnotatedString.Builder()
+                builder.append(normalizeNewlines(getTextContent(appContext, v.data.text)))
+                builder.toAnnotatedString()
+            } else if (v.data.hasStyledText()) {
+                val builder = AnnotatedString.Builder()
+                for (run in getTextContent(appContext, v.data.styledText)) {
+                    val style =
+                        run.styleOrNull
+                            ?: throw NoSuchFieldException(
+                                "Malformed data: StyledTextRun's style unset"
+                            )
```

Comment:
I think we can remove these `orNull ?: throw exception` bits. These fields are meant to be required, so we don't need to "dereference" the Java Optional type. If we did want to do some checking to make sure that required fields are set, then I think it'd be best to do so in a centralized set of functions.
```suggestion
                        run.style
```

Reply:
I agree with making them centralized if possible. I'm not sure we need these either, but perhaps they'd be useful for debugging purposes. I haven't used this to really know if this might be helpful or not.

Reply:
Done